app/models/concerns/applicant/nir.rb

```diff
@@ -3,22 +3,33 @@ module Applicant::Nir
 
   included do
     validate :nir_is_valid, if: :nir?
+    after_validation :add_nir_key
   end
 
   private
 
+  def add_nir_key
+    self.nir = "#{nir}#{nir_key}" if nir&.length == 13
```

Comment:
Ça revient au même mais j'ai l'impression que c'est un tt petit peu plus lisible en écrivant comme ça (j'enlève au passe le safe operator vu ma suggestion précédente)
```suggestion
    return unless nir.length == 13
    
    self.nir = "#{nir}#{nir_key}" 
```